File: src/main/java/com/innercircle/survey/common/exception/CustomException.java
package com.innercircle.survey.common.exception;

import lombok.Getter;

@Getter
public class CustomException extends RuntimeException {

    private final ErrorCode errorCode;

    public CustomException() {
        this.errorCode = ErrorCode.INTERNAL_SERVER_ERROR;
    }

    public CustomException(ErrorCode errorCode) {
        super(errorCode.getMessage());
        this.errorCode = errorCode;
    }
}

File: survey-common/src/main/kotlin/com/innercircle/surveycommon/exception/CustomException.kt
package com.innercircle.surveycommon.exception

sealed class CustomException(
    message: String,
    val errorCode: String,
) : RuntimeException(message)

class ResourceNotFoundException(
    message: String = "리소스를 찾을 수 없습니다",
    errorCode: String = "RESOURCE_NOT_FOUND",
) : CustomException(message, errorCode)

class InvalidInputException(
    message: String = "잘못된 입력입니다",
    errorCode: String = "INVALID_INPUT",
) : CustomException(message, errorCode)
